client/components/Appointment.jsx

```diff
@@ -0,0 +1,186 @@
+    };
+    defaultTimes();
+
+    const formatTimes = (opens, closes) => {
+      const mutableOpens = moment(opens, 'HH:mm:a');
+      const timesArray = [mutableOpens.format('LT')];
+      for (let j = 0; j < 12; j += 1) {
+        const endHour = moment(closes, 'HH:mm:a');
+        const currHour = moment(timesArray[timesArray.length - 1], 'HH:mm:a').add(1, 'h');
+        if (endHour.isBefore(moment('04:00:00', 'HH:mm:a')) && currHour.diff(endHour) <= 79200000) {
+          timesArray.push(currHour.format('LT'));
+        } else if (currHour.isBefore(endHour.subtract(0.5, 'h'))) {
+          timesArray.push(currHour.format('LT'));
+        }
+      }
+      const defaultTime = timesArray[0];
+      this.setState({
+        timeRange: timesArray,
+        selectedTime: defaultTime,
+      });
+    };
+
+    const { businessId } = this.state;
+    $.ajax({
+      url: `/${businessId}`,
+      method: 'GET',
+      dataType: 'json',
+      success: (dbData) => {
+        this.setState({
+          businessName: dbData[0].name,
+          businessMax: dbData[0].guest_max,
+          businessAddress: dbData[0].address,
+          businessCity: dbData[0].city,
+          businessState: dbData[0].state,
+          businessZip: dbData[0].zip,
+        });
+        formatTimes(dbData[0].opens, dbData[0].closes);
+      },
+    });
+  }
+
+  next() {
+    const { view } = this.state;
+    this.setState({ view: view + 1 });
+  }
+
+  back() {
+    const { view } = this.state;
+    this.setState({ view: view - 1 });
+  }
+
+  handleChange(e) {
+    const field = e.target.name;
+    const target = e.target.value;
+    this.setState({ [field]: target });
+  }
+
+  makeAppointment() {
+    const {
+      firstName, lastName, phone, email,
+      businessId, selectedTime, selectedDate, guestCount,
+    } = this.state;
+
+    $.ajax({
+      url: `/${businessId}/appointment`,
+      method: 'POST',
+      data: {
+        userDetails: {
+          firstName,
+          lastName,
+          phone,
+          email,
+        },
+        appointmentDetails: {
+          businessId,
+          customerId: null,
+          time: selectedTime,
+          date: selectedDate,
+          count: guestCount,
+        },
+      },
+      dataType: 'json',
+    });
+    this.forceUpdate();
+  }
+
+  render() {
+    const {
+      nextTwoWeeks, timeRange, businessMax, view,
+      businessName, businessAddress, businessCity, businessState, businessZip,
+      selectedDate, selectedTime, firstName, lastName, email, phone, guestCount,
+    } = this.state;
+
+    if (view === 1) {
+      return (
+        <div>
+          <LevelOne
+            nextTwoWeeks={nextTwoWeeks}
+            times={timeRange}
+            max={businessMax}
+            change={this.handleChange}
+            next={this.next}
+            count={guestCount}
+            date={selectedDate}
+            time={selectedTime}
+          />
+        </div>
+      );
+    }
+    if (view === 2) {
```

Comment:
I see that next and back add / subtract from view.  Do you have it set up so that there's no way to get outside the bounds of [1, 2]?  Otherwise you might want to check for valid view values.

Reply:
This control is built in, but in a less code intensive way. Next button only appears on view 1, while back only appears on view 2.